xarray/core/computation.py

```diff
@@ -579,28 +597,65 @@ def apply_variable_ufunc(
                 "``.load()`` or ``.compute()``"
             )
         elif dask == "parallelized":
-            input_dims = [broadcast_dims + dims for dims in signature.input_core_dims]
             numpy_func = func
 
+            if dask_gufunc_kwargs is None:
+                dask_gufunc_kwargs = {}
+
+            output_sizes = dask_gufunc_kwargs.pop("output_sizes", dict())
+            if output_sizes:
+                output_sizes_renamed = dict()
+                for key, value in output_sizes.items():
+                    if key not in signature.all_output_core_dims:
+                        raise ValueError(
+                            f"dimension name '{key}' in 'output_sizes' must correspond to output_core_dims"
+                        )
+                    output_sizes_renamed[signature.dims_map[key]] = value
+                dask_gufunc_kwargs["output_sizes"] = output_sizes_renamed
+
+            for key in signature.all_output_core_dims:
+                if key not in signature.all_input_core_dims:
+                    if key not in output_sizes:
+                        raise ValueError(
+                            f"dimension name '{key}' in 'output_core_dims' needs corresponding (key,value) in 'output_sizes'"
+                        )
+
             def func(*arrays):
-                return _apply_blockwise(
+                import dask.array as da
+
+                res = da.apply_gufunc(
                     numpy_func,
-                    arrays,
-                    input_dims,
-                    output_dims,
-                    signature,
-                    output_dtypes,
-                    output_sizes,
-                    meta,
+                    signature.to_gufunc_string(),
+                    *arrays,
+                    **dask_gufunc_kwargs,
                 )
 
+                # todo: covers for https://github.com/dask/dask/pull/6207
+                #  remove when minimal dask version >= 2.17.0
+                from dask import __version__ as dask_version
+
+                if LooseVersion(dask_version) < LooseVersion("2.17.0"):
+                    if signature.num_outputs > 1:
+                        res = tuple(r for r in res)
+
+                return res
+
         elif dask == "allowed":
             pass
         else:
             raise ValueError(
                 "unknown setting for dask array handling in "
                 "apply_ufunc: {}".format(dask)
             )
+    else:
+        if dask_gufunc_kwargs is not None:
```

Comment:
I think this is part of Stephan's earlier point.

It's confusing that we are looking for `vectorize` and `output_dtypes` in `dask_gufunc_kwargs` when we are not using `apply_gufunc`. I think we should  pass these two kwargs explicitly in for clarity.

Reply:
done